crates/polars-plan/src/plans/optimizer/simplify_functions.rs

```diff
@@ -7,6 +7,58 @@ pub(super) fn optimize_functions(
     expr_arena: &mut Arena<AExpr>,
 ) -> PolarsResult<Option<AExpr>> {
     let out = match function {
+        // is_null().any() -> null_count() > 0
+        // is_not_null().any() ->  null_count() < len()
+        FunctionExpr::Boolean(BooleanFunction::Any { ignore_nulls: _ }) => {
```

Comment:
nit: leave a comment above that documents we can ignore the value of the `ignore_nulls` parameter as `is_null()/is_not_null()` are guaranteed not to produce NULLs